include/esp_can_update.h

```diff
@@ -111,45 +110,50 @@ class CANUpdate
                     update_progress_message_.Enable();
                 }
             }
-            else if (update_started_ && message_type_ == MessageType::kUpdateData)
+        },
+        message_type_,
+        length_signal_group_,
+        md5_signal_group_};
+
+    CANRXMessage<2> update_data_message_{
+        can_interface_,
+        kUpdateId,
+        [this]() {
+            uint64_t data = update_data_;
+            uint32_t index = ((update_data_message_.GetID() & 0x1FFFF800) >> 3) + data_block_index_low_;
```

Comment:
Where does the 0x1FFFF800 come from?  Can you add comment or give value descriptive name.